include/zephyr/drivers/video.h

```diff
@@ -202,6 +202,220 @@ struct video_frmival_enum {
 	};
 };
 
+/**
+ * @name Port and Endpoint DT Helpers
+ * @{
+ */
+
+#define _DT_INST_PORT_BY_ID(inst, pid)                                                             \
+	COND_CODE_1(DT_NODE_EXISTS(DT_INST_CHILD(inst, ports)),                                    \
+	    (DT_CHILD(DT_INST_CHILD(inst, ports), port_##pid)), (DT_INST_CHILD(inst, port_##pid)))
+
+/**
+ * @brief Get a port node from its id
+ *
+ * Given a device instance number, return a port node specified by its ID.
+ * It handles various ways of how a port could be defined. For example:
+ *
+ * @code{.c}
+ *	DT_PROP(DT_INST_PORT_BY_ID(inst, 0), property)
+ * @endcode
+ *
+ * This above snippet will return @c "port_property" for each @c &device listed below:
+ *
+ * @code{.dts}
+ *	&device {
+ *		ports {
+ *			port@0 {
+ *				reg = <0x0>;
+ *				property = "port_property";
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		port@0 {
+ *			reg = <0x0>;
+ *			property = "port_property";
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		port {
+ *			property = "port_property";
+ *		};
+ *	};
+ * @endcode
+ *
+ * @param inst instance number
+ * @param pid port ID
+ * @return port node associated with @p pid
+ */
+#define DT_INST_PORT_BY_ID(inst, pid)                                                              \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_PORT_BY_ID(inst, pid)),                                \
+		(_DT_INST_PORT_BY_ID(inst, pid)), (DT_INST_CHILD(inst, port)))
+
+#define _DT_INST_ENDPOINT_BY_ID(inst, pid, eid)                                                    \
```

Comment:
I *think* these still need to be documented, right? Even if the documentation is simply "Helper for DT_INST_PORT_BY_ID"

Reply:
done